server-common/src/main/java/org/apache/kafka/server/config/ServerLogConfigs.java

```diff
@@ -178,6 +178,12 @@ public class ServerLogConfigs {
     public static final String ALTER_CONFIG_POLICY_CLASS_NAME_CONFIG = "alter.config.policy.class.name";
     public static final String ALTER_CONFIG_POLICY_CLASS_NAME_DOC = "The alter configs policy class that should be used for validation. The class should " +
             "implement the <code>org.apache.kafka.server.policy.AlterConfigPolicy</code> interface.";
+    public static final String ALTER_CONFIG_POLICY_KRAFT_COMPATIBILITY_ENABLE_CONFIG = "alter.config.policy.kraft.compatibility.enable";
```

Comment:
Keeping this configuration will cause an exception when upgrading to 4.x with the existing configuration file. We could make the 4.x broker ignore this configuration, or add a note to `upgrade.md` reminding users to remove it